Fix bug in AOTI lowering

tugsbayasgalan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #148488
* #148485
* #148483
* __->__ #148364



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

Fixes: https://github.com/pytorch/pytorch/issues/148370

Differential Revision: [D70514480](https://our.internmc.facebook.com/intern/diff/D70514480)

tugsbayasgalan committed:
Fix bug in AOTI lowering

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D70514480).

tugsbayasgalan committed:
Update on "Fix bug in AOTI lowering"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 amjames chauhang aakhundov

Fixes: https://github.com/pytorch/pytorch/issues/148370

Differential Revision: [D70514480](https://our.internmc.facebook.com/intern/diff/D70514480)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D70514480).

tugsbayasgalan commented:
@pytorchbot label "topic: not user facing"

desertfire commented on test/inductor/test_aot_inductor.py:
```diff
@@ -1162,6 +1162,21 @@ def forward(self, q, k, v):
 
+    def test_aoti_constant_tensor(self):
+        class Foo(torch.nn.Module):
+            def __init__(self):
```
You need to specify a device here to cover both cpu and cuda, like here https://github.com/pytorch/pytorch/blob/a41413829c98377e4c155ff150f250438303f7b2/test/inductor/test_aot_inductor.py#L210 .

desertfire commented on test/inductor/test_aot_inductor.py:
```diff
@@ -1162,6 +1162,21 @@ def forward(self, q, k, v):
+                return torch.ops.aten.linear.default(x, self.a, self.b)
+
+        ep = torch.export.export(
+            Foo(), (torch.ones(4, 4),), strict=False
```
The same when you generating example inputs.

desertfire commented on torch/_inductor/compile_fx.py:
```diff
@@ -661,9 +661,9 @@ def _compile_fx_inner(
-    )
+    assert isinstance(
+        next(iter(reversed(gm.graph.nodes))).args[0], (tuple, list)
+    ), f"inductor can only compile FX graphs which return a tuple/list, but got {gm.graph}"
```
Did you install the lastest version of lintrunner? Why is there a difference here?

tugsbayasgalan committed:
Update on "Fix bug in AOTI lowering"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 amjames chauhang aakhundov

Fixes: https://github.com/pytorch/pytorch/issues/148370

Differential Revision: [D70514480](https://our.internmc.facebook.com/intern/diff/D70514480)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D70514480).

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D70514480).

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D70514480).

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D70514480).
